### DISCUSSION TITLE
Who is actually implementing and running Tests for Pimcore Projects

### DISCUSSION THREAD

**Discussion by dpfaffenbauer** [pimcore/pimcore#13176] [Polls]
Do you implement tests and run them in CI before deploying code with Pimcore?

**Comment by yariksheptykin** [comment]
"Yes" with PHP-Unit or codeception depending on the project setup. My preferred option is PHPUnit + Symfony Test classes.

**Comment by dpfaffenbauer** [comment]
ok, wow, not a lot of people do tests with pimcore....
